flink-ml-iteration/src/main/java/org/apache/flink/ml/iteration/Iterations.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.iteration;
+
+import org.apache.flink.annotation.PublicEvolving;
+
+/** A helper class to create iterations. */
+@PublicEvolving
+public class Iterations {
+
```

Comment:
Hmm... it seems that the FLIP uses the class name `IterationUtils`. We will need to update the FLIP first if we agree to use this name.

I would slightly prefer `IterationUtils` because this class provides just static public methods. But I am not strong on this. Can you explain the rational for `Iterations`? We can also ask other reviewers for opinions.

Reply:
It seems to me `IterationUtils` seems more like some utility methods used inside the iteration instead of a factory to create iterations? The name `Iterations` followed some convention of the flink's name to the factory class like `RestartStrategies`.